sdk/src/main/java/io/opentelemetry/sdk/metrics/DoubleObserverSdk.java

```diff
@@ -33,9 +42,20 @@
         monotonic);
   }
 
+  @Override
+  List<MetricData> collectAll() {
+    Callback<ResultDoubleObserver> currentMetricUpdater = metricUpdater;
+    if (currentMetricUpdater == null) {
+      return Collections.emptyList();
+    }
+    final ActiveBatcher activeBatcher = getActiveBatcher();
+    currentMetricUpdater.update(new ResultDoubleObserverSdk(activeBatcher, isMonotonic()));
+    return activeBatcher.completeCollectionCycle();
+  }
+
   @Override
   public void setCallback(Callback<DoubleObserver.ResultDoubleObserver> metricUpdater) {
-    throw new UnsupportedOperationException("to be implemented");
+    this.metricUpdater = Utils.checkNotNull(metricUpdater, "metricUpdater");
```

Comment:
Since you check for `null` values in `collectAll()`, I would drop this requirement. This way, we can remove a callback after it is set

Reply:
Not sure if that is the behavior I am looking for, I would like to start more restrictive than open the requirements instead of the other way around (backwards incompatible change).

Reply:
Thx for the explanation! I now agree with you :smile: